lib/router/router.js

```diff
@@ -769,7 +769,21 @@ function updateURL(transition, state /*, inputUrl*/) {
     var isQueryParamsRefreshTransition =
       transition.queryParamsOnly && urlMethod === 'replace';
 
-    if (initial || replaceAndNotAborting || isQueryParamsRefreshTransition) {
+    // say you are at / and you a `replaceWith(/foo)` is called. Then, that
+    // transition is aborted with `replaceWith(/bar)`. At the end, we should
+    // end up with /bar replacing /. We are replacing the replace. We only
+    // will replace the initial route if all subsequent aborts are also
+    // replaces. However, there is some ambiguity around the correct behavior
```

Comment:
Can you describe the ambiguity you are referring to here? (I'd like to make sure we are all on the same page RE: the factors and assumptions)

Reply:
The ambiguity I see is a replace followed by a series that includes an update. It's not clear to me if that should add a history entry or not.